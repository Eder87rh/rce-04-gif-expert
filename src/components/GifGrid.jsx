import React, { Fragment, useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  useEffect(() => {
    getGifs(category).then(setImages);
  }, [category]);

  return (
    <Fragment>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image}>
            {image.title}
          </GifGridItem>
        ))}
      </div>
    </Fragment>
  );
};

export default GifGrid;
