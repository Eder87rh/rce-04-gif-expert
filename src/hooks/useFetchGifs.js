import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((images) => {
      console.log(
        "ğŸš€ ~ file: useFetshGifs.js ~ line 13 ~ useEffect ~ images",
        images
      );
      setState({
        data: images,
        loading: false,
      });
    });
  }, [category]);

  return state;
};
