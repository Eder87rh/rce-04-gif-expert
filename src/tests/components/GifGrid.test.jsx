import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid', () => {

  it('should load correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category="Category" />);
    expect(wrapper).toMatchSnapshot();
  });  

  it('should show items when load images useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/dasdasdasd.jpg',
      title: 'sdadasdasd'
    }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category="Category" />);
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  
})
