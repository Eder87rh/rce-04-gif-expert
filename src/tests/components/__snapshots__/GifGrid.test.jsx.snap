// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should load correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
